### PR TITLE
allow list_transpose() to work on data frames

### PR DESCRIPTION
**Pull request by KimLopezGuell**
Fixed issue #1109

### CODE REVIEW

File: tests/testthat/test-list-transpose.R [hunk 1]
```diff
@@ -140,3 +140,8 @@ test_that("fail mixing named and unnamed vectors", {
     test_list_transpose()
   })
 })
+
+test_that("works on data frames", {
+  expect_no_error(list_transpose(mtcars))
```
Comment (krlmlr) [inline tests/testthat/test-list-transpose.R:145]: Could this be a test with a simple data frame or tibble as in the test above?

```r
x <- tibble(list(a = 1, b = 2), list(a = 3, b = 4))
...
```

**Review by krlmlr — COMMENTED** [review]
Thanks, looks great!

Do you want to update the documentation of the `x` argument in the roxygen2 header above? Could you please also add a bullet to `NEWS.md` ?

File: R/list-transpose.R [hunk 2]
```diff
@@ -69,7 +69,13 @@ list_transpose <- function(x,
                            simplify = NA,
                            ptype = NULL,
                            default = NULL) {
-  vec_check_list(x)
+  if(!is.list(x)) {
```
Comment (hadley) [inline R/list-transpose.R:72]: Looks like purrr already had a `check_list()` so I used it here.

**Comment by hadley** [comment]
Thanks for working on this! I hope you had fun at TDD 😁

**Pull request merged**